uisdk/src/main/java/tech/dojo/pay/uisdk/domain/mapper/PaymentIntentDomainEntityMapper.kt

```diff
@@ -3,22 +3,57 @@ package tech.dojo.pay.uisdk.domain.mapper
 import tech.dojo.pay.uisdk.domain.entities.PaymentIntentDomainEntity
 import tech.dojo.pay.uisdk.domain.entities.PaymentIntentStatusDomainEntity
+import java.util.Currency
 
 internal class PaymentIntentDomainEntityMapper {
-    fun apply(raw: PaymentIntentPayload): PaymentIntentDomainEntity {
-        checkInvalidParameters(raw)
-        return PaymentIntentDomainEntity(
+    fun apply(raw: PaymentIntentPayload): PaymentIntentDomainEntity? {
+        return if (containsInvalidParameters(raw)) {
+            null
+        } else {
+            mapToPaymentIntentDomainEntityWithValidRaw(raw)
+        }
+    }
+
+    private fun containsInvalidParameters(raw: PaymentIntentPayload): Boolean {
+        val invalidParams: MutableList<String> = mutableListOf()
+        if (raw.id == null) invalidParams.add("id")
+        if (raw.clientSessionSecret == null) invalidParams.add("clientSessionSecret")
+        if (raw.amount == null && raw.intendedAmount == null) invalidParams.add("amount")
+        if (!isValidCurrencyCode(raw.amount?.currencyCode ?: raw.intendedAmount?.currencyCode)) invalidParams.add("currencyCode")
+        if (raw.merchantConfig == null) invalidParams.add("merchantConfig")
+        if (raw.merchantConfig?.supportedPaymentMethods == null) invalidParams.add("supportedPaymentMethods")
+        if (raw.merchantConfig?.supportedPaymentMethods?.cardSchemes == null) invalidParams.add("cardSchemes")
+        if (raw.status == null) invalidParams.add("status")
+        return invalidParams.isNotEmpty()
+    }
+
+    private fun isValidCurrencyCode(currencyCode: String?): Boolean {
+        return currencyCode?.let {
+            try {
+                Currency.getInstance(it) != null
+            } catch (e: IllegalArgumentException) {
+                false
+            }
+        } ?: false
+    }
+
+    private fun mapToPaymentIntentDomainEntityWithValidRaw(raw: PaymentIntentPayload) =
+        PaymentIntentDomainEntity(
             id = requireNotNull(raw.id),
             customerId = raw.customer?.id,
             paymentToken = requireNotNull(raw.clientSessionSecret),
             amount = AmountDomainEntity(
                 valueLong = requireNotNull(raw.amount?.value ?: raw.intendedAmount?.value),
-                valueString = requireNotNull(raw.amount?.value?.centsToString() ?: raw.intendedAmount?.value?.centsToString()),
-                currencyCode = requireNotNull(raw.amount?.currencyCode ?: raw.intendedAmount?.currencyCode),
+                valueString = requireNotNull(
+                    raw.amount?.value?.centsToString()
+                        ?: raw.intendedAmount?.value?.centsToString(),
+                ),
+                currencyCode = requireNotNull(
+                    raw.amount?.currencyCode ?: raw.intendedAmount?.currencyCode,
+                ),
             ),
             supportedCardsSchemes = requireNotNull(raw.merchantConfig?.supportedPaymentMethods?.cardSchemes?.mapNotNull { it }),
```

Comment:
`requireNotNull` is still here, can you remove them? I know that you already checked before, but if for any case you forget to check a parameter then the code will break.
For strings add `.orEmpty` and for lists `?: emptyList()`